talaiot/src/main/kotlin/com/cdsap/talaiot/configuration/MetricsConfiguration.kt

```diff
@@ -49,12 +50,24 @@ import com.cdsap.talaiot.metrics.base.Metric
  *  [GradleSwitchRerunTasksMetric]
  *  [GradleSwitchDaemonMetric]
  *
- *  If you want to define a custom metrics:
+ *  If you want to define custom metrics:
  *
+ *  ```
  *  metrics {
- *    customBuildMetrics["key"] = "value"
- *    customTaskMetrics["key"] = "value"
+ *    // Custom build metrics
+ *    customBuildMetrics(
+ *      "metricKey" to "metricValue"
+ *    )
+ *    // Custom task metrics
+ *    customTaskMetrics(
+ *      "metricKey" to "metricValue"
+ *    )
+ *    // Custom metric implementation
+ *    customMetrics(
+ *      MyCustomMetric()
+ *    )
  *  }
+ *  ```
  */
 class MetricsConfiguration {
```

Comment:
Perfect. 
Just one question, in the description of the PR you mention:

> Previously, even if you set both build or task metrics, the defaults would be added.
 This behaviour goes against the comment in the source code that states "If there was any customization then we assume that user-populated everything manually".
> The code is changed to add both custom build and custom task metrics directly into metrics as SimpleMetrics. This is done before build() is called, so there's a small change in its behaviour.


I want to clarify that if we are not adding metrics or specific Talaiot metrics, still we will get:
```
if (metrics.isEmpty()) {
  default()
  performance()
  gradleSwitches()
  git()
  environment()
}
```

Reply:
Yes, that is correct. You will get the defaults if no custom metrics are defined. :+1: 
Before you would get the defaults, even if you added custom metrics, because they were stored in a separate map and executed after the `metrics.isEmpty()` check.